docs/virtual_machines/liveness_and_readiness_probes.md

```diff
@@ -243,3 +250,107 @@ watchdog -t 2000ms -T 4000ms /dev/watchdog
 the watchdog will send a heartbeat every two seconds to `/dev/watchdog` and
 after four seconds without a heartbeat the defined action will be executed. In
 this case a hard `poweroff`.
+
+## Defining Guest-Agent Ping Probes
+
+A Guest-Agent probes are based on qemu-guest-agent `guest-ping`.
+This will ping the guest and return an error if the guest is not
+up and running.  To easily define this on VM spec, specify `guestAgentPing: {}`
+in VM's readiness probe spec; `virt-controller` will translate this into a
+corresponding command wrapped by `virt-probe`.
+
+> Note: You can only define one of the type of probe.
+
+
+**Important:** If the qemu-guest-agent is not installed **and** enabled inside
+the VM, the probe will fail.  Many images don't enabled the agent by default so
+make sure you either run one that does or enable it. 
+
+Make sure to provide enough delay and failureThreshold for the VM and the agent
+to be online.
+
+In the following example the Fedora image does have qemu-guest-agent available
+by default. Nevertheless, in case qemu-guest-agent is not installed, it will be
+installed and enabled via cloud-init as shown in the example below.  Also,
+cloud-init assigns the proper SELinux context, i.e. virt_qemu_ga_exec_t, to the
+`/tmp/healthy.txt` file.  Otherwise, SELinux will deny the attempts to open the
+`/tmp/healthy.txt` file causing the probe to fail.
+
+```yaml
```

Comment:
Comparing to other examples in this page, yours is the only one I see using VM and not VMI. Would you be open to changing to VMI for consistency (unless this would miss a feature you need)?

Reply:
You are right. Changed by a VMI example.